app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace App\Http\Controllers;
+
+use App\Http\Requests\PostRequest;
+use App\Models\Category;
+use App\Models\Post;
+use App\Services\PostService;
+
+class PostController extends Controller
+{
+    public function __construct(PostService $postService)
+    {
+        $this->postService = $postService;
+    }
+
+    public function index()
+    { 
+        $this->viewData['posts'] = $this->postService->listPost();
+
+        return view('admin.post.index', $this->viewData);
+    }
+    
+    public function create()
+    {
+        $category = Category::orderBy('created_at', 'DESC')->get();
+
+        return view('admin.post.create')->with(compact('category'));
+    }
+
+    public function store(PostRequest $request)
+    {
+        $this->postService->create($request->all());
+
+        return back()->with('success', 'Thêm thành công');
+    }
+
+    public function edit(Post $post, Category $category)
+    {
+        $category = Category::orderBy('created_at', 'DESC')->get();
+
+        return view('admin.post.edit',compact('post'), compact('category'));
```

Comment:
em có thể sử dụng kiểu `this->viewData[categories] = .... `
`this->viewData[post] = .... ` rồi gửi viewData qua nhé, như các function bên trên ấy